### PR TITLE
feat: Handling serialization of immutablelist in configuration/settings

### PR DESCRIPTION
**Pull request by nickrandolph**
GitHub Issue (If applicable): #659

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

UpdateAsync has a non-immutable option for updating settings
ImmutableList is not supported

## What is the new behavior?

The non-immutable overload has been removed
Immutabelist can be used in settings/configuration entity

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.Extensions.Navigation.UI/Navigators/TypeExtensions.cs [hunk 1]
```diff
@@ -6,12 +6,11 @@ public static class TypeExtensions
 {
 	public static ConstructorInfo? GetNavigationConstructor(this Type type, INavigator navigator, IServiceProvider services, out object[] constructorArguments)
 	{
-		var ctrs = type.GetConstructors();
-		foreach (var ctr in ctrs)
+		var ctr = type.GetConstructors().FirstOrDefault();
```
Comment (dr1rrb) [inline src/Uno.Extensions.Navigation.UI/Navigators/TypeExtensions.cs:9]: Should we be able to configure which ctor to use?

File: src/Uno.Extensions.Configuration/ConfigurationBinder.cs [hunk 2]
```diff
@@ -0,0 +1,731 @@
...
+	}
+
+	private static object BindImmutableList(
+		object collection,
+			Type collectionType,
+		IConfiguration config, BinderOptions options)
+	{
+		// ICollection<T> is guaranteed to have exactly one parameter
+		Type itemType = collectionType.GenericTypeArguments[0];
+		MethodInfo addMethod = collectionType.GetMethod("Add", DeclaredOnlyLookup);
```
Comment (dr1rrb) [inline src/Uno.Extensions.Configuration/ConfigurationBinder.cs:518]: Should be great to use the `Builder` when available (NTH - the real fix would be to use the `System.Text.Json`).

File: src/Uno.Extensions.Configuration/ConfigurationBinder.cs [hunk 3]
```diff
@@ -0,0 +1,731 @@
+﻿
+
+using System.Collections.Immutable;
+using System.ComponentModel;
+
+namespace Uno.Extensions.Configuration.Internal;
+
+public static class ConfigurationBinder
```
Comment (dr1rrb) [inline src/Uno.Extensions.Configuration/ConfigurationBinder.cs:8]: Indeed LOT of reflection here ... would be great to replace that by `System.Text.Json` (in an other PR)

**Review by dr1rrb — APPROVED** [review]

File: src/Uno.Extensions.Navigation.UI/Navigators/TypeExtensions.cs (hunk 1 above)
Comment (nickrandolph) [reply src/Uno.Extensions.Navigation.UI/Navigators/TypeExtensions.cs:9]: I can't see a strong use case for supporting multiple constructors for view model creation. In the case where different data can be passed into the same view model type, you could have multiple optional parameters on the constructor

File: src/Uno.Extensions.Configuration/ConfigurationBinder.cs (hunk 3 above)
Comment (nickrandolph) [reply src/Uno.Extensions.Configuration/ConfigurationBinder.cs:8]: Yeh, there's a trade off between converting configuration section to a string and then parsing using System.Text.Json versus the current logic that uses reflection. Loading configuration isn't done that frequently, and the conversion to strongly typed entity (ie this binder logic) isn't done immediately on startup, only when the entities are actually requested

**Pull request merged**